extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/TracerRecorder.java

```diff
@@ -31,19 +35,40 @@
     }
 
     /* STATIC INIT */
-    public RuntimeValue<SdkTracerProvider> createTracerProvider(String serviceName, String serviceVersion,
+    public RuntimeValue<SdkTracerProvider> createTracerProvider(TracerConfig config,
+            String serviceName,
+            String serviceVersion,
             ShutdownContext shutdownContext) {
         BeanManager beanManager = Arc.container().beanManager();
+
+        Instance<IdGenerator> idGenerator = beanManager.createInstance()
+                .select(IdGenerator.class, Any.Literal.INSTANCE);
+
         SdkTracerProviderBuilder builder = SdkTracerProvider.builder();
 
+        // Define ID Generator if present
+        if (idGenerator.isResolvable()) {
+            builder.setIdGenerator(idGenerator.get());
+        }
+
+        DelayedAttributes delayedAttributes = beanManager.createInstance()
+                .select(DelayedAttributes.class, Any.Literal.INSTANCE).get();
+
+        delayedAttributes.setAttributesDelegate(Resource.getDefault()
```

Comment:
Are we also adding here the processing of any `Resource` CDI Beans into the `Resource` so the late bound are only config/env properties?